llvm/lib/CodeGen/MachineTraceMetrics.cpp

```diff
@@ -546,7 +540,7 @@ MachineTraceMetrics::Ensemble::invalidate(const MachineBasicBlock *BadMBB) {
   if (BadTBI.hasValidHeight()) {
     BadTBI.invalidateHeight();
     WorkList.push_back(BadMBB);
-    do {
+    while (!WorkList.empty()) {
```

Comment:
is this change NFC? The loop used to execute a minimum of 1 time. Now it executes a minimum of 0 times.

Reply:
See next line: `Worklist.pop_back_val`, which would fail on an empty worklist.

Reply:
I missed the `WorkList.push_back(BadMBB);`. We know the worklist will never start out empty. So this change is good.